resources/repository.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+#
+# Cookbook:: postgresql
+# Resource:: repository
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+property :version, String, default: '10'
+property :minor_version, String, default: '10-1'
+property :enable_pgdg, [true,false], default: false
+property :enable_pgdg_source, [true,false], default: false
+property :enable_pgdg_updates_testing, [true,false], default: true
+property :enable_pgdg_source_updates_testing, [true,false], default: false
+
+default_action :add
+
+action :add do
+  case node['platform_family']
+
+  when 'rhel'
+    remote_file "/etc/pki/rpm-gpg/RPM-GPG-KEY-PGDG-#{new_resource.version}" do
+      source "https://download.postgresql.org/pub/repos/yum/RPM-GPG-KEY-PGDG-#{new_resource.version}"
+    end
+
+    yum_repository "PostgreSQL #{new_resource.version} $releasever - $basearch" do
```

Comment:
It seems like instead of the 3 different properties we could make a property called yum_repos (or whatever) and pass an array. That array would create the various repos and default to the non-source repos.

Reply:
There's only one here? Unless I'm mistaken.